private/buf/cmd/buf/command/push/push.go

```diff
@@ -347,6 +388,306 @@ func validateLabelFlagValues(flags *flags) error {
 	return nil
 }
 
+// We do not allow users to set --source-control-url, --create-default-label, and --label
+// flags if the --git-metadata flag is set.
+func validateGitMetadataFlags(flags *flags) error {
+	if flags.GitMetadata {
+		var usedFlags []string
+		if flags.SourceControlURL != "" {
```

Comment:
Not blocking but I feel like if we couldn't detect a source code URL, we should allow the user to specify one (or fail push with an error). Otherwise they might not realize that git-metadata is failing to discover information.

Alternatively, we could warn if we fail to detect a URL or a HEAD branch (if --create is specified).

Additionally, labels should be additive - so we'll add labels by default for git metadata but it may be that users want to specify additional labels on commits and we should just take the intersection of them both.

Reply:
> if we couldn't detect a source code URL, we should allow the user to specify one

I think we should fail if we can't detect a source control URL or default branch.

> we should just take the intersection of them both

We discussed this and decided that, for now, it's one or the other. `--git-metadata` opines that Git is the source of truth for labels. If there are more labels to associate, associate them in Git and we'll push them. If you need something outside of that, you can't use `--git-metadata`.

Reply:
> Additionally, labels should be additive - so we'll add labels by default for git metadata but it may be that users want to specify additional labels on commits and we should just take the intersection of them both.

Yes, I had a conversation with folks this morning, and we're definitely allowing users to add `--label`s to this.

For now, we are disallowing `--source-control-url` and `--create-default-label`, but this may change.